src/content/blog/astro-4120.mdx

```diff
@@ -0,0 +1,140 @@
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Server Islands
+
+Today we're announcing experimental support for **Server Islands**, bringing islands architecture to the server. With Server Islands you mark components to run dynamically on the server during page load, allowing the rest of the page to be cached at the edge.
+
+In 2021 Astro brought the idea of [islands architecture](https://astro.build/blog/introducing-astro/) to the mainstream. We noticed that developers were building content oriented sites using frontend frameworks even though little of the page were dynamic. Islands architecture provides a way to limit the amount of JavaScript loaded in the client without sacrificing the DX (developer experience) that modern frameworks afford.
+
+With **Server Islands** we are extending this same architecture to the server side as well. Within any given web page you might have content that is:
+
+- Completely static and never changes.
+- Dynamically backed by a database that changes infrequently, but more often than you deploy.
+- Personalized content, tailored to individual users.
+
+Today you must choose 1 caching strategy for all of these types of content, and if the page is a logged-in experience that usually means no caching at all. Server Islands allow you to mark certain components as being islands. Doing so will **defer** them from loading until the browser has first downloaded the shell. 
+
+<BlogContentImage src={islandDiag} alt="A diagram showing the server island population parts of the page from the server." />
+
+The server islands are used for your most dynamic content; personalized content like the user's avatar, their shopping cart, and product reviews. With these components deferred you're free to more aggressively cache the page itself.
+
+This means that users, logged in or not, will see the most critical parts of the page instantly, as they are caching on Edge CDNs. With careful design considerations the fallback content will be visible for a short amount of time before the dynamic islands are then loaded.
+
+Each island is loaded independent from the rest; this means a slower island, such as one connected to a legacy backend, won't delay the rest of personalized content from being seen and interacted with.
+
+### Trying out Server Islands
+
+We're still working on polishing the feature, but you can try out Astro Server Islands today by enabling the feature in Astro 4.12:
+
+```js
+import { defineConfig } from 'astro/config';
+import netlify from '@astrojs/netlify';
+
+export default defineConfig({
+  output: 'hybrid',
+  adapter: netlify(),
+
+  experimental: {
+    serverIslands: true,
+  },
+})
+```
+
+Server islands, as the name implies, run on the server, so you'll need to use either `'server'` or `'hybrid'` output. Once you've enabled the feature you can use them on any component using the `server:defer` directive:
```

Comment:
With this sentence I understood: you can use `server:defer` and `client:load` at the same time

```astro
<Foo server:defer client:load />
```
Is that what you wanted to say?